src/main/java/school/lemon/changerequest/java/io/hw1/FileManager.java

```diff
@@ -0,0 +1,175 @@
+        try {
+            File file = new File(name);
+            if (file.exists()) throw new FileAlreadyExistException(file);
+            file.createNewFile();
+            System.out.println("File has been created.");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void createDirectory(String name) throws DirectoryAlreadyExistException, FileManagerException {
+        File directory = new File(name);
+        if (directory.exists() && directory.isDirectory()) throw new DirectoryAlreadyExistException(directory);
+        try {
+            if (directory.mkdir()) System.out.println("Directory has been created. " + directory.getAbsolutePath());
+        } catch (FileManagerException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void changeDirectory(String name) throws IllegalDirectoryException {
+        File newCurrentDirectory = new File(name).getAbsoluteFile();
+        if (!newCurrentDirectory.exists() || !newCurrentDirectory.isDirectory())
+            throw new IllegalDirectoryException(newCurrentDirectory);
+        currentDirectory = newCurrentDirectory;
+        System.out.println("Current directory has been changed.");
+    }
+
+    @Override
+    public boolean delete(String name) throws FileNotFoundException {
+        boolean delete = delete(name, true);
+        return delete;
+    }
+
+    @Override
+    public boolean delete(String name, boolean force) throws FileNotFoundException {
+        File fileToDelete = new File(name).getAbsoluteFile();
+        if (!fileToDelete.exists()) throw new FileNotFoundException(name);
+        if (fileToDelete.isFile() || fileToDelete.list().length == 0) {
+            fileToDelete.delete();
+            return true;
+        } else {
+            if (force == false) return false;
+            File[] files = fileToDelete.listFiles();
+            for (File f : files) {
+                f.delete();
+            }
+            if (fileToDelete.list().length == 0) {
+                fileToDelete.delete();
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void copy(String source, String dest, CopyMode copyMode)
+            throws FileNotFoundException, IllegalArgumentException, FileAlreadyExistException, FileCopyException {
+        File fileSource = new File(source).getAbsoluteFile();
+        if (!fileSource.exists()) throw new FileNotFoundException(source);
+        if (fileSource.isDirectory()) throw new IllegalArgumentException(source);
+        File fileDest = new File(dest);
+        if (fileDest.exists()) throw new FileAlreadyExistException(fileDest);
+        try {
+            if (copyMode.equals(CopyMode.SIMPLE)) {
+                Files.copy(fileSource.toPath(), fileDest.toPath());
+                System.out.println("Copied successfully.");
+            } else {
+                InputStream is = null;
```

Comment:
No need to split declaration and initialization here. So it can be `InputStream is = new FileInputStream(fileSource);`